controller/engine_controller.go

```diff
@@ -1768,6 +1770,11 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replica, addr st
 			// it is past the Backoff period, we'll try to mark the Replica as Failed and increase the Backoff period
 			// for the next failure.
 			if !ec.backoff.IsInBackOffSinceUpdate(e.Name, time.Now()) {
+				if err := updateReplicaRebuildFailedCondition(log, ec.ds, replica, replicaRebuildErrMsg); err != nil {
+					log.WithError(err).Errorf("Failed to update rebuild status information on replica %v", replica)
+					return
+				}
```

Comment:
Suggest

```go
if err := updateReplicaRebuildFailedCondition(log, ec.ds, rep, replicaRebuildErrMsg); err != nil {
    log.WithError(err).Errorf("Failed to update rebuild status information on replica %v", replica)
    return
}
```

and move the below code to the above and outer scope, so you don't need to repeat querying the replica CR..

```go
rep, err := ec.ds.GetReplica(replica)
if err != nil {
    log.WithError(err).Errorf("Failed to get replica %v unable to mark failed rebuild", replica)
    return
}
```

Reply:
Modified. And Move `GetReplica` out of the goroutine function. Thanks